code/modules/research/xenobiology/crossbreeding/reproductive.dm

```diff
@@ -3,46 +3,63 @@ Reproductive extracts:
 	var/extract_type = /obj/item/slime_extract/
-	var/cubes_eaten = 0
+	var/cooldown = 3 SECONDS
+	var/feedAmount = 3
 	var/last_produce = 0
-	var/cooldown = 30 // 3 seconds.
+	var/datum/component/storage/concrete/extract_inventory/slimeStorage
+
+/obj/item/slimecross/reproductive/examine()
+	. = ..()
+	. += span_danger("It appears to have eaten [length(contents)] Monkey Cube[p_s()]")
+
+/obj/item/slimecross/reproductive/Initialize()
+	. = ..()
+	slimeStorage = AddComponent(/datum/component/storage/concrete/extract_inventory)
 
 /obj/item/slimecross/reproductive/attackby(obj/item/O, mob/user)
 	if((last_produce + cooldown) > world.time)
 		to_chat(user, span_warning("[src] is still digesting!"))
 		return
+
+	if(length(contents) >= feedAmount) //if for some reason the contents are full, but it didnt digest, attempt to digest again
+		to_chat(user, span_warning("[src] appears to be full but is not digesting! Maybe poking it stimulated it to digest."))
+		slimeStorage.processCubes(src, user)
+		return
+
 	if(istype(O, /obj/item/storage/bag/bio))
 		var/list/inserted = list()
-		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 1, null, null, user, inserted)
+		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, feedAmount - length(contents), TRUE, FALSE, user, inserted)
 		if(inserted.len)
-			var/obj/item/food/monkeycube/M = inserted[1]
-			if(istype(M))
-				eat_cube(M, user)
+			to_chat(user, span_notice("You feed [length(inserted)] Monkey Cube[p_s()] to [src], and it pulses gently."))
+			playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
+			slimeStorage.processCubes(src, user)
 		else
 			to_chat(user, span_warning("There are no monkey cubes in the bio bag!"))
-	if(istype(O,/obj/item/food/monkeycube))
-		eat_cube(O, user)
-	if(cubes_eaten >= 3)
-		var/cores = rand(1,4)
-		visible_message(span_notice("[src] briefly swells to a massive size, and expels [cores] extract[cores > 1 ? "s":""]!"))
-		playsound(src, 'sound/effects/splat.ogg', 40, TRUE)
-		last_produce = world.time
-		for(var/i = 0, i < cores, i++)
-			new extract_type(get_turf(loc))
-		cubes_eaten = 0
-
-/obj/item/slimecross/reproductive/proc/eat_cube(obj/item/food/monkeycube, mob/user)
-		qdel(monkeycube)
-		cubes_eaten++
-		to_chat(user, span_notice("You feed [monkeycube] to [src], and it pulses gently."))
+		return
+
+	else if(istype(O, /obj/item/food/monkeycube))
+		slimeStorage.locked = FALSE //This weird unlock-then-lock nonsense brought to you courtesy of storage jank
+		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_INSERT, O, user, TRUE, FALSE)
+		slimeStorage.locked = TRUE
+		to_chat(user, span_notice("You feed 1 Monkey Cube to [src], and it pulses gently."))
+		slimeStorage.processCubes(src, user)
 		playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
+		return
+	else
+		to_chat(user, span_notice("The [src] rejects the Monkey Cube!")) //in case it fails to insert for whatever reason you get feedback
```

Comment:
```suggestion
	else if(istype(O, /obj/item/food/monkeycube))
		slimeStorage.locked = FALSE //This weird unlock-then-lock nonsense brought to you courtesy of storage jank
		if(SEND_SIGNAL(src, COMSIG_TRY_STORAGE_INSERT, O, user, TRUE))
			to_chat(user, span_notice("You feed 1 Monkey Cube to [src], and it pulses gently."))
			slimeStorage.processCubes(src, user)
			playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
			slimeStorage.locked = TRUE //relock once its done inserting
			return
		else
			slimeStorage.locked = TRUE //it couldnt insert for some reason, relock it
			to_chat(user, span_notice("The [src] rejects the Monkey Cube!")) //in case it fails to insert for whatever reason you get feedback
```

the final else is in the wrong area, you want to do an if() check for the signal (which doesnt need the final FALSE, its false by default), if the signal actually works it returns a true, which does the code under it, if the signal didnt work, it returns false, which moves onto the else statement which gives you a message about its failure, in both cases it will relock the storage once its done